powertools-parameters/powertools-parameters-ssm/src/main/java/software/amazon/lambda/powertools/parameters/ssm/SSMProviderBuilder.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package software.amazon.lambda.powertools.parameters.ssm;
+
+import software.amazon.awssdk.core.SdkSystemSetting;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.core.client.config.SdkAdvancedClientOption;
+import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.ssm.SsmClient;
+import software.amazon.lambda.powertools.common.internal.UserAgentConfigurator;
+import software.amazon.lambda.powertools.parameters.BaseProvider;
+import software.amazon.lambda.powertools.parameters.cache.CacheManager;
+import software.amazon.lambda.powertools.parameters.transform.TransformationManager;
+
+public class SSMProviderBuilder {
+    private SsmClient client;
+    private CacheManager cacheManager;
+    private TransformationManager transformationManager;
+
+    private static SsmClient createClient() {
+        return SsmClient.builder()
+                .httpClientBuilder(UrlConnectionHttpClient.builder())
+                .region(Region.of(System.getenv(SdkSystemSetting.AWS_REGION.environmentVariable())))
+                .overrideConfiguration(ClientOverrideConfiguration.builder()
+                        .putAdvancedOption(SdkAdvancedClientOption.USER_AGENT_SUFFIX,
+                                UserAgentConfigurator.getUserAgent(BaseProvider.PARAMETERS)).build())
+                .build();
+    }
+
+    /**
+     * Create a {@link SSMProvider} instance.
+     *
+     * @return a {@link SSMProvider}
+     */
+    public SSMProvider build() {
+        if (cacheManager == null) {
+            // TODO - do we want to share this somehow?
```

Comment:
do we? or do we remove the TODO ?
what do you mean by sharing the cacheManager ?